core/src/main/java/io/fabric8/maven/core/util/SpringBootUtil.java

```diff
@@ -80,25 +101,82 @@ protected static Properties getPropertiesResource(URL resource) {
+        return getPropertiesFromYamlResource(resource, activeProfiles);
     }
 
     /**
      * Returns a {@code Properties} representation of the given Yaml resource or an empty properties object if the resource is null
      */
-    protected static Properties getPropertiesFromYamlResource(URL resource) {
+    protected static Properties getPropertiesFromYamlResource(URL resource, List<String> activeProfiles) {
         if (resource != null) {
             try (InputStream yamlStream = resource.openStream()) {
-                Yaml yaml = new Yaml();
-                @SuppressWarnings("unchecked")
-                SortedMap<String, Object> source = yaml.loadAs(yamlStream, SortedMap.class);
+                Yaml yaml = new Yaml(new SafeConstructor());
+
+                Map<String, Map> profileDocs = new HashMap<>();
+
                 Properties properties = new Properties();
-                if (source != null) {
-                    properties.putAll(getFlattenedMap(source));
+
+                Iterable<Object> yamlDoc = yaml.loadAll(yamlStream);
+
+                Iterator yamlDocIterator = yamlDoc.iterator();
+
+                int docCount = 0;
+                while (yamlDocIterator.hasNext()) {
+                    Map docRoot = (Map) yamlDocIterator.next();
+
+                    String profiles = null;
+
+                    if (docRoot.containsKey("spring")) {
+
+                        LinkedHashMap value = (LinkedHashMap) docRoot.get("spring");
+
+                        Object profilesValue = value.get("profiles");
+
+                        if (profilesValue instanceof Map) {
+                            Map profileMap = (Map) profilesValue;
+                            String[] arrayOfActiveProfiles =
+                                    StringUtils.split((String) profileMap.get("active"), ",");
+                            if (activeProfiles.isEmpty() && docCount > 0 && arrayOfActiveProfiles != null) {
+                                for (int i = 0; i < arrayOfActiveProfiles.length; i++) {
+                                    String activeProfile = arrayOfActiveProfiles[i];
+                                    activeProfiles.add(StringUtils.trim(activeProfile));
+                                }
+                            }
+                        } else if (profilesValue instanceof String) {
+                            profiles = (String) profilesValue;
+                        }
+                    }
+
+                    if (profiles != null) {
+                        String[] profileSplit = profiles.split("\\s*,\\s*");
+                        if (!CollectionUtils.
+                                intersection(Arrays.asList(profileSplit), activeProfiles)
+                                .isEmpty()) {
+                            //if the profiles is in the list of active profiles we add it to our list of docs
+                            profileDocs.put(profiles, docRoot);
```

Comment:
no thats taken care by previous block, this in case where the yaml has only one doc so ideally we will have only one profile.. i hope my understanding is right